src/parse_stream.jl

```diff
@@ -155,10 +155,27 @@ const NO_POSITION = ParseStreamPosition(0, 0)
 
 #-------------------------------------------------------------------------------
 """
-ParseStream provides an IO interface for the parser. It
-- Wraps the lexer with a lookahead buffer
-- Removes insignificant whitespace and comment tokens, shifting them into the
-  output implicitly (newlines may be significant depending on `skip_newlines`)
+    ParseStream(text::AbstractString,          index::Integer=1; version=VERSION)
+    ParseStream(text::IO;                                        version=VERSION)
+    ParseStream(text::Vector{UInt8},           index::Integer=1; version=VERSION)
+    ParseStream(ptr::Ptr{UInt8}, len::Integer, index::Integer=1; version=VERSION)
+
+Construct a `ParseStream` from source `text` which may come in various forms -
+a string, an `IO` object, or a buffer of bytes. In the case that the buffer is
```

Comment:
I don't think we have an official `IO` interface. Do you think it'd make sense to list the requirements somewhere (in particular, `seek`ability comes to mind)?

Reply:
Yes it makes sense. I thought I'd done this, but it's in the docs for `parse!()` I'll put it here too